jmix-flowui/flowui/src/main/java/io/jmix/flowui/app/bulk/BulkEditView.java

```diff
@@ -0,0 +1,586 @@
+                        }
+                    }
+                    EntityValues.setValueEx(item, fieldEntry.getKey(), value);
+                }
+            }
+        }
+    }
+
+    protected void ensureEmbeddedPropertyCreated(E item, String propertyPath) {
+        if (!StringUtils.contains(propertyPath, ".")) {
+            return;
+        }
+
+        MetaPropertyPath path = context.getMetaClass().getPropertyPath(propertyPath);
+
+        if (path != null) {
+            Object currentItem = item;
+            for (MetaProperty property : path.getMetaProperties()) {
+                if (metadataTools.isEmbedded(property)) {
+                    Object currentItemValue = EntityValues.getValue(currentItem, property.getName());
+                    if (currentItemValue == null) {
+                        Object newItem = metadata.create(property.getRange().asClass());
+                        EntityValues.setValue(currentItem, property.getName(), newItem);
+                        currentItem = newItem;
+                    } else {
+                        currentItem = currentItemValue;
+                    }
+                } else {
+                    break;
+                }
+            }
+        }
+    }
+
+    protected void showConfirmDialogOrCommit(List<String> fields) {
+        if (context.isUseConfirmDialog()) {
+            dialogs.createOptionDialog()
+                    .withHeader(messageBundle.getMessage("bulk.confirmation"))
+                    .withText(messageBundle.formatMessage("bulk.applyConfirmation",
+                            items.size(), StringUtils.join(fields, "\n")))
+                    .withActions(new DialogAction(Type.OK)
+                                    .withText(messages.getMessage("actions.Apply"))
+                                    .withHandler(event -> commitBulkChanges()),
+                            new DialogAction(Type.CANCEL)
+                                    .withVariant(ActionVariant.PRIMARY)
+                                    .withHandler(e -> loadItems()))
+                    .open();
+        } else {
+            commitBulkChanges();
+        }
+    }
+
+    protected void commitBulkChanges() {
+        List<String> fields = new ArrayList<>();
+
+        for (Map.Entry<String, AbstractField<?, ?>> fieldEntry : dataFields.entrySet()) {
+            AbstractField<?, ?> field = fieldEntry.getValue();
+            if (isFieldChanged(field)) {
+                fields.add(managedFields.get(fieldEntry.getKey()).getFqn());
+            }
+        }
+
+        EntitySet saved = dataManager.save(new SaveContext().saving(items));
+
+        Logger logger = LoggerFactory.getLogger(BulkEditView.class);
+        logger.info("Applied bulk editing for {} entries of {}. Changed properties: {}",
+                saved.size(), context.getMetaClass(), StringUtils.join(fields, ", "));
+
+        notifications.create(messageBundle.formatMessage("bulk.successMessage", saved.size()))
```

Comment:
This message should have the same style as success save message for entities, i.e. `success` type and located at the top end